riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideRegistrar.java

```diff
@@ -138,27 +140,29 @@ private String registerBaseURL(final String id, final Client client) {
         });
     }
 
-    private Object createExecutor(final String id, final Client client) {
-        if (client.getThreads().getEnabled()) {
-            return ref(registerExecutor(id, client));
+    private Object createExecutor(final String id, final String metricName, final Client client, @Nullable final RiptideProperties.Threads threads) {
+        if (threads != null && threads.getEnabled()) {
```

Comment:
How threads can become null?

Reply:
By default, Failsafe policies use `null` for threads (unlike client HTTP threads), so I’ve added a null check to handle that case safely.